### PR TITLE
[JIT] add support for torch.cdist

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33818 [JIT] Add support for list()
* #33783 [JIT] add support for torch.norm
* **#33737 [JIT] add support for torch.cdist**
* #33736 [JIT] add support for lu_unpack
* #33724 [JIT] add support for torch.lu to torchscript
* #33504 [JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft

Differential Revision: [D20121916](https://our.internmc.facebook.com/intern/diff/D20121916)

### CODE REVIEW

**Review by vishwakftw — APPROVED** [review]
Looks good to me, thank you.

**Comment by vishwakftw** [comment]
Are the docs failures normal?

**Comment by eellison** [comment]
> Are the docs failures normal?

i think i just need to rename `indices_product` to `_indices_product` the PR below this

File: torch/jit/_builtins.py [hunk 1]
```diff
@@ -83,7 +83,8 @@
     (torch.nn.init._no_grad_zero_, "aten::_no_grad_zero_"),
     (torch._C._get_tracing_state, "aten::_get_tracing_state"),
     (warnings.warn, "aten::warn"),
-    (torch._VF.stft, "aten::stft")
+    (torch._VF.stft, "aten::stft"),
```
[the remark this answers is not in the document]
Comment (eellison) [reply torch/jit/_builtins.py]: i think it's something to do with native schema bound ops name collision.

**Pull request closed without merging**